### PR TITLE
Create CI/PR maintenance jobs, migrate clean-ssh-keys

### PR DESCRIPTION
**Pull request by fejta**
Closes https://github.com/kubernetes/test-infra/issues/908

Add unit test coverage for job-templates:
* bootstrap projects use job-templates defined in the same file
* bootstrap job-templates are always consumed by a project

Add CI jobs to testgrid

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/test-infra/910)

### CODE REVIEW

**Comment by fejta** [comment]
@krzyzacy @ixdy ping!

File: jobs/maintenance-pull-clean-ssh-keys.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  k8s-jkns-pr-gke
+  k8s-jkns-pr-gci-gke
+  k8s-jkns-pr-gci-kubemark
+  k8s-jkns-pr-bldr-e2e-gce-fdrtn
+  k8s-jkns-pr-gce-etcd3
+  k8s-jkns-pr-gci-bld-e2e-gce-fd
+  )
+
+for proj in "${projects[@]}"; do
+  echo ./jenkins/clean_project.py --project="${proj}" --hours=3 --delete
```
Comment (krzyzacy) [inline jobs/maintenance-pull-clean-ssh-keys.sh]: isn't this script kills all the instances, why the job name is clean-ssh-keys?

File: jenkins/job-configs/kubernetes-jenkins-pull/bootstrap-maintenance-pull.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    - shell: |
+        # TODO(fejta): consider a stable tag instead of master
+        git clone https://github.com/kubernetes/test-infra -b master
+        './test-infra/jenkins/bootstrap.py' --job='{job-name}' --repo='{repo-name}' --branch='{branch}' --root="${{GOPATH}}/src"
+    concurrent: true
+    properties:
+    - build-discarder:
+        days-to-keep: 30
+    - throttle:
+        max-total: 10
```
Comment (ixdy) [inline jenkins/job-configs/kubernetes-jenkins-pull/bootstrap-maintenance-pull.yaml]: I'm not sure we want to run 10 of these at once?

File: jenkins/job-configs/kubernetes-jenkins-pull/bootstrap-maintenance-pull.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    name: 'maintenance-pull-{suffix}'  # Runs on each node once
+    builders:
+    - shell: |
+        # TODO(fejta): consider a stable tag instead of master
+        git clone https://github.com/kubernetes/test-infra -b master
+        './test-infra/jenkins/bootstrap.py' --job='{job-name}' --repo='{repo-name}' --branch='{branch}' --root="${{GOPATH}}/src"
+    concurrent: true
+    properties:
+    - build-discarder:
+        days-to-keep: 30
```
Comment (ixdy) [inline jenkins/job-configs/kubernetes-jenkins-pull/bootstrap-maintenance-pull.yaml]: previously this was set to 14 days, I think.

File: jenkins/job-configs/kubernetes-jenkins/bootstrap-maintenance-ci.yaml [hunk 4]
```diff
@@ -0,0 +1,43 @@
+- job-template:
+    name: 'maintenance-ci-{suffix}'  # Runs on each node once
```
Comment (ixdy) [inline jenkins/job-configs/kubernetes-jenkins/bootstrap-maintenance-ci.yaml]: need to set a node for CI Jenkins.

comments for PR Jenkins also apply here.

File: jenkins/bootstrap_test.py [hunk 5]
```diff
@@ -732,6 +732,26 @@ def Check(job, name):
 
         self.CheckBootstrapYaml('job-configs/bootstrap-maintenance.yaml', Check)
 
+    def testBootstrapMaintenanceCIYaml(self):
+        def Check(job, name):
+            job_name = 'maintenance-ci-%s' % name
+            self.assertEquals(job_name, job.get('job-name'))
+            self.assertIn('frequency', job)
+            self.assertIn('repo-name', job)
+            self.assertIn('.', job['repo-name'])  # Has domain
```
Comment (fejta) [inline jenkins/bootstrap_test.py]: Factored some of this out (although I generally try to minimize abstractions in tests. I generally try to optimize the ease of understanding over the number of lines.

File: jenkins/job-configs/kubernetes-jenkins-pull/bootstrap-maintenance-pull.yaml (hunk 3 above)
Comment (fejta) [inline jenkins/job-configs/kubernetes-jenkins-pull/bootstrap-maintenance-pull.yaml]: Any concerns with this being 30 and/or something special about 14?

File: jenkins/job-configs/kubernetes-jenkins-pull/bootstrap-maintenance-pull.yaml (hunk 2 above)
Comment (fejta) [inline jenkins/job-configs/kubernetes-jenkins-pull/bootstrap-maintenance-pull.yaml]: Done

File: jenkins/job-configs/kubernetes-jenkins/bootstrap-maintenance-ci.yaml (hunk 4 above)
Comment (fejta) [inline jenkins/job-configs/kubernetes-jenkins/bootstrap-maintenance-ci.yaml]: Done

File: jobs/maintenance-pull-clean-ssh-keys.sh (hunk 1 above)
Comment (fejta) [inline jobs/maintenance-pull-clean-ssh-keys.sh]: Done, doh!

**Pull request merged**